transforms/helpers/validation-helper.ts

```diff
@@ -71,7 +73,8 @@ export function hasValidProps(
       }
 
       if (
-        instanceProp.type === 'ObjectExpression' &&
+        (instanceProp.type === 'ObjectExpression' ||
+          instanceProp.type === 'ArrayExpression') &&
         instanceProp.name !== 'queryParams'
```

Comment:
Whoa, weird hardcoded exception! I know it predates you, but weird!

Reply:
IDK. It sorta makes sense when you look at https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/avoid-leaking-state-in-ember-objects.md